src/hotspot/share/opto/library_call.cpp

```diff
@@ -5192,6 +5195,80 @@ void LibraryCallKit::create_new_uncommon_trap(CallStaticJavaNode* uncommon_trap_
   uncommon_trap_call->set_req(0, top()); // not used anymore, kill it
 }
 
+//------------------------------inline_array_partition-----------------------
+bool LibraryCallKit::inline_array_partition() {
+
+  address stubAddr = nullptr;
+  const char *stubName;
+  stubName = "array_partition_stub";
+
+  Node* elementType     = null_check(argument(0));
+  Node* obj             = argument(1);
+  Node* offset          = argument(2);
+  Node* fromIndex       = argument(4);
+  Node* toIndex         = argument(5);
+  Node* pivot_indices   = argument(6);
+  Node* pivot_offset    = argument(7);
+  Node* isDualPivot     = argument(9);
+
+  const TypeInstPtr* elem_klass = gvn().type(elementType)->isa_instptr();
+  ciType* elem_type = elem_klass->const_oop()->as_instance()->java_mirror_type();
+  BasicType bt = elem_type->basic_type();
+  stubAddr = StubRoutines::select_array_partition_function(bt);
+  if (stubAddr == nullptr) return false;
```

Comment:
I see now how you check for AVX512 support.

You bailout here if address for stubs is not set and I see that you have `if (UseAVX > 2 && VM_Version::supports_avx512dq())` check in stubGenerator.